src/objective-c/GRPCClient/private/GRPCHost.m

```diff
@@ -78,6 +78,7 @@ - (instancetype)initWithAddress:(NSString *)address {
     if (cachedHost) {
       return cachedHost;
     }
+  }
```

Comment:
This   @synchronized block isn't correct. @synchronized only works if both the only reader and the only writer of hostCache are inside the block.

As written, the only writer, line ~86, is outside the @synchronized which leads to a non-theread-safe race condition